### DISCUSSION TITLE
How to add PWA ??

### DISCUSSION THREAD

**Discussion by wahyusa** [filamentphp/filament#1856] [Help] [answered]
Hi, thanks for this great package !!
I have done to add manifest.json/webmanifest in meta head by filament
But I don't know how to add my custom javascript like a basic check for navigator etc maybe like this in all pages/layout.

```javascript
if ('serviceWorker' in navigator) {
  // Use the window load event to keep the page load performant
  window.addEventListener('load', () => {
    navigator.serviceWorker.register('/service-worker.js');
  });
}
```

I have also try to use [Workbox laravel mix](https://laravel-mix.com/extensions/workbox) and add script in app.js but still not work.

if I miss something please let me know...

**Comment by danharrin** [comment, marked as answer]
You need to [register the script](https://filamentphp.com/docs/2.x/admin/appearance#including-frontend-assets) in a service provider.

**Reply by wahyusa** [reply]
Thanks! It works.
In my case don't forget to add crossorigin attribute like this if you found 401 Unauthorized to manifest.json

```html
<link rel="manifest" crossorigin="use-credentials" href="/manifest.json">
```
